packages/components/src/components/hds/app-side-nav/portal/target.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { DEBUG } from '@glimmer/env';
+import { macroCondition, isTesting } from '@embroider/macros';
+
+import type { HdsAppSideNavPortalSignature } from './index';
+
+// import { PortalTargetSignature } from 'ember-stargate/components/portal-target';
+interface PortalTargetSignature {
+  Element: HTMLDivElement;
+  Args: {
+    name: string;
+    multiple?: boolean;
+    onChange?: (count: number) => void;
+  };
+  Blocks: {
+    default: [number];
+  };
+}
+
+import type { Registry as Services } from '@ember/service';
+
+interface HdsAppSideNavPortalTargetSignature {
+  Args: PortalTargetSignature['Args'] & {
+    targetName?: HdsAppSideNavPortalSignature['Args']['targetName'];
+  };
+  Element: HTMLDivElement;
+}
+
+export default class HdsAppSideNavPortalTarget extends Component<HdsAppSideNavPortalTargetSignature> {
+  @service router!: Services['router'];
+
+  @tracked numSubnavs = 0;
+  @tracked lastPanelEl: Element | undefined;
+
+  static get prefersReducedMotionOverride(): boolean {
+    return macroCondition(isTesting()) ? true : false;
+  }
+
+  prefersReducedMotionMQ = window.matchMedia(
+    '(prefers-reduced-motion: reduce)'
+  );
+
+  get prefersReducedMotion(): boolean {
+    return (
+      HdsAppSideNavPortalTarget.prefersReducedMotionOverride ||
+      (this.prefersReducedMotionMQ && this.prefersReducedMotionMQ.matches)
+    );
+  }
+
+  @action
+  panelsChanged(portalCount: number): void {
+    this.numSubnavs = portalCount;
+  }
+
+  @action
+  didUpdateSubnav(element: HTMLElement, [count]: [number]): void {
+    this.animateSubnav(element, [count]);
+  }
+
+  @action
+  animateSubnav(element: HTMLElement, [count]: [number]): void {
+    /*
+     * Here is what the layout looks like for this setup
```

Comment:
Please add to the comment that this is ascii art. See https://www.w3.org/TR/WCAG20-TECHS/H86.html